src/neural/xla/onnx2hlo.cc

```diff
@@ -458,13 +587,87 @@ class Onnx2HloConverter {
                             is_sorted, is_unique)};
   }
 
+  HloFlow DoReduceMean(HloFlow input, const std::vector<int64_t>& axes) {
+    HloFlow zero = MakeScalar(0, input->shape().element_type());
+    auto flow = builder_.Reduce(
+        input, zero, MakeAddComputation(input->shape().element_type()), axes);
+    size_t count = std::accumulate(
+        axes.begin(), axes.end(), 1, [&](size_t acc, size_t axis) {
+          return acc * input->shape().dimensions(axis);
+        });
+    auto denominator =
+        DoBroadcast(MakeScalar(count, input->shape().element_type()),
+                    flow->shape().dimensions());
+    flow = builder_.Divide(flow, denominator);
+    HloTensorType target_shape(input->shape());
+    for (auto axis : axes) target_shape.SetDimension(axis, 1);
+    return builder_.Reshape(flow, target_shape);
+  }
+
+  std::vector<HloFlow> OpReduceMean(const pblczero::NodeProto& node) {
+    CheckKnownAttributes(node, 1, {"axes"});
+    auto* input = GetInput(node, 0);
+    auto axes = GetAttribute(node, "axes")->ints();
+    return {DoReduceMean(input, axes)};
+  }
+
+  std::vector<HloFlow> OpCast(const pblczero::NodeProto& node) {
+    CheckKnownAttributes(node, 1, {"to"});
+    auto* input = GetInput(node, 0);
+    const auto onnx_type = static_cast<pblczero::TensorProto::DataType>(
+        GetAttribute(node, "to")->i());
+    const auto hlo_type = OnnxTypeToXlaType(onnx_type);
+    if (input->shape().element_type() == hlo_type) return {input};
+    return {builder_.Convert(input, hlo_type)};
+  }
+
+  std::vector<HloFlow> OpLayerNormalization(const pblczero::NodeProto& node) {
```

Comment:
`LayerNormalization` was added with opset 17, but onnxruntime accepts it with earlier versions as well, so probably no need for a version check  here.